src/main/kotlin/su/arlet/business1/security/SecurityConfig.kt

```diff
@@ -0,0 +1,114 @@
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.security.access.hierarchicalroles.RoleHierarchy
+import org.springframework.security.access.hierarchicalroles.RoleHierarchyImpl
+import org.springframework.security.authentication.AuthenticationManager
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider
+import org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration
+import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity
+import org.springframework.security.config.annotation.web.builders.HttpSecurity
+import org.springframework.security.config.annotation.web.configurers.CorsConfigurer
+import org.springframework.security.config.annotation.web.configurers.CsrfConfigurer
+import org.springframework.security.config.http.SessionCreationPolicy
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
+import org.springframework.security.crypto.password.PasswordEncoder
+import org.springframework.security.web.SecurityFilterChain
+import org.springframework.security.web.access.expression.DefaultWebSecurityExpressionHandler
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter
+import su.arlet.business1.security.jwt.AuthEntryPoint
+import su.arlet.business1.security.jwt.AuthTokenFilter
+import su.arlet.business1.security.sevices.AuthUsersDetailsService
+
+
+@Configuration
+@SecurityScheme(
+    name = "Authorization",
+    type = SecuritySchemeType.HTTP,
+    bearerFormat = "JWT",
+    scheme = "bearer"
+)
+@EnableMethodSecurity(prePostEnabled = true)
+class SecurityConfig @Autowired constructor(
+    private val userDetailsService: AuthUsersDetailsService,
+    private val authEntryPoint: AuthEntryPoint
+) {
+    @Value("\${api.path}")
+    private var apiPath = ""
+
+    @Bean
+    fun authenticationJwtTokenFilter(): AuthTokenFilter {
+        return AuthTokenFilter()
+    }
+
+    @Bean
+    fun authenticationProvider(): DaoAuthenticationProvider {
+        val authProvider = DaoAuthenticationProvider()
+        authProvider.setUserDetailsService(userDetailsService)
+        authProvider.setPasswordEncoder(passwordEncoder())
+        return authProvider
+    }
+
+    @Bean
+    @Throws(Exception::class)
+    fun authenticationManager(authConfig: AuthenticationConfiguration): AuthenticationManager {
+        return authConfig.getAuthenticationManager()
+    }
+
+    @Bean
+    fun passwordEncoder(): PasswordEncoder {
+        return BCryptPasswordEncoder()
+    }
+
+    @Bean
+    fun roleHierarchy(): RoleHierarchy {
+        val roleHierarchy = RoleHierarchyImpl()
+        roleHierarchy.setHierarchy(
+            "ROLE_ADMIN > ROLE_EDITOR\n" +
```

Comment:
Can change \n to """ or something like that. I forget

Reply:
Not needed